packages/ui-kit/src/Breadcrumbs/hooks/useModelHook.tsx

```diff
@@ -0,0 +1,82 @@
+import { useState, useLayoutEffect, useRef } from 'react';
+
+interface UseModelProps<T> {
+  items: T[];
+  ellipsisWidth: number;
+}
+
+export const useModel = <T,>({ items, ellipsisWidth }: UseModelProps<T>) => {
+  const [itemWidths, setItemWidths] = useState<number[]>([]);
+  const [leftHiddenItemIndex, setLeftHiddenItemIndex] = useState(-1);
+  const [rightHiddenItemIndex, setRightHiddenItemIndex] = useState(items.length);
+  const olRef = useRef<HTMLOListElement>(null);
+
+  // Cache all item widths before the first render
+  useLayoutEffect(() => {
+    if (!olRef.current) return;
+
+    const widths = Array.from(olRef.current.children).map((child) => (child as HTMLElement).offsetWidth);
+    setItemWidths(widths);
+  }, [items]);
+
+  useLayoutEffect(() => {
+    const calculateBreadcrumbs = () => {
+      if (!olRef.current || itemWidths.length === 0) return;
+
+      const olComputedStyle = window.getComputedStyle(olRef.current);
+      const paddingLeft = parseFloat(olComputedStyle.paddingLeft);
+      const paddingRight = parseFloat(olComputedStyle.paddingRight);
+      const containerWidth = olRef.current.clientWidth - paddingLeft - paddingRight;
+
+      const totalWidth = itemWidths.reduce((acc, width) => acc + width, 0);
+
+      if (totalWidth <= containerWidth) {
+        // No overflow, display all items
+        setLeftHiddenItemIndex(-1);
```

Comment:
Maybe it would be easier and more obvious to check for `null`, instead of [-1; items.length]?
Plus, this interval kinda breaks the logic of "all items between two indexes need to be hidden"